lib/semver.js

```diff
@@ -161,6 +161,35 @@ exports.checkValid = (version) => {
   return true;
 };
 
+/**
+ * @summary Compare extended semver versions. Used for sorting
+ * @function
+ * @private
+ *
+ * @param {String} a - extended semver string
+ * @param {String} b - extended semver string
+ * @returns {Number}
+ * a > b => 1
+ * a = b => 0
+ * a < b => -1
+ *
+ * @example
+ * compareExtendedSemver('2.1.1', '2.1.0')
+ * > 1
+ * compareExtendedSemver('2.1.1', '2.1.0+rev1')
+ * > 1
+ * compareExtendedSemver('2.1.1+rev1', '2.1.1+rev2')
+ * > -1
```

Comment:
How does it behave when doing e.g. `compareExtendedSemver('2.1.1', '2.1.1+rev1')`

Reply:
This would still be -1

Reply:
Added a test case for that